course6/toyapp-adam/util/keyboard.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <cstdint>
+#include <SDL2/SDL.h>
+
+namespace Util {
+
+class Keyboard
+{
+public:
+	using KeyCode = SDL_Keycode;
+
+	void Press(KeyCode keyCode);
+	void Release(KeyCode keyCode);
+	void ClearChanges();
+	bool IsUp(KeyCode keyCode) const;
+	bool IsDown(KeyCode keyCode) const;
+	bool IsPressed(KeyCode keyCode) const;
+	bool IsReleased(KeyCode keyCode) const;
+private:
+	struct KeyState
+	{
+		std::uint8_t up : 1;
```

Comment:
I don't get it. What's he difference between ```up``` and ```released``` and ```down``` and ```pressed```?
By the look of it, you could just store one boolean per key: ```isPressed```?